test/t/conftest.py

```diff
@@ -755,16 +755,22 @@ def assert_complete(
         bash.send(cmd + "\t")
         # Sleep a bit if requested, to avoid `.*` matching too early
         time.sleep(kwargs.get("sleep_after_tab", 0))
-        bash.expect_exact(kwargs.get("rendered_cmd", cmd))
+        rendered_cmd = kwargs.get("rendered_cmd", cmd)
+        bash.expect_exact(rendered_cmd)
         bash.send(MAGIC_MARK)
         got = bash.expect(
             [
                 # 0: multiple lines, result in .before
-                r"\r\n" + re.escape(PS1 + cmd) + ".*" + re.escape(MAGIC_MARK),
+                r"\r\n"
+                + re.escape(PS1 + rendered_cmd)
```

Comment:
This switches from expecting `cmd` to expecting `rendered_cmd` here, but the commit message doesn't mention it. It may very well be the right thing to do, but I think we should mention if/that the change is deliberate as now it's in a commit that only mentions checking error messages which doesn't explain it to the casual reader.

Could you add the elaboration on this? Also the explanations you're adding to PR descriptions for each commit are great, but they would be better placed in the actual commit message bodies.

Reply:
OK. I think I'll separate the commit and also add commit messages. Also, it seems this change found a few existing errors of `nounset`, so I'll add their fixes.